workload.go

```diff
@@ -62,32 +76,59 @@ func (w WorkloadConfig) Config() WorkloadConfig {
 }
 
 func NewWorkloadConfigWithDefaults(c WorkloadConfig) WorkloadConfig {
-	if c.RateControl.OuterLoopRate == 0 {
-		c.RateControl.OuterLoopRate = 50
+	if c.Name == "" {
+		panic("must specify workload Name")
 	}
 
-	if c.Visualization.LatencyHistMin == 0 {
-		c.Visualization.LatencyHistMin = 0
+	if c.WorkloadScale <= 0 || c.WorkloadScale >= 1 {
+		panic("WorkloadScale (if specified) must be greater than 0.0 and less than 1.0")
 	}
 
-	if c.Visualization.LatencyHistMax == 0 {
-		c.Visualization.LatencyHistMax = 50000
+	if c.RateControlConfig.EventRate == 0 {
+		c.RateControlConfig.EventRate = 1000
 	}
 
-	if c.Visualization.LatencyHistSize == 0 {
-		c.Visualization.LatencyHistSize = 1000
+	if c.RateControlConfig.Concurrency == 0 {
+		if c.RateControlConfig.MaxEventRatePerWorker == 0 {
+			c.RateControlConfig.MaxEventRatePerWorker = 100
+		}
+		c.RateControlConfig.Concurrency = int(math.Ceil(c.RateControlConfig.EventRate / float64(c.RateControlConfig.MaxEventRatePerWorker)))
+	} else if c.RateControlConfig.Connections >= c.RateControlConfig.Concurrency {
+		c.DatabaseConfig.ConnectionMultiplier = c.RateControlConfig.Connections / c.RateControlConfig.Concurrency
+	} else {
+		c.DatabaseConfig.ConnectionMultiplier = 1
 	}
 
-	if c.RateControl.MaxEventRatePerGoroutine == 0 {
-		c.RateControl.MaxEventRatePerGoroutine = 100 // 2 Event per iteration by default.
+	if c.RateControlConfig.MaxEventRatePerWorker == 0 {
+		c.RateControlConfig.MaxEventRatePerWorker = math.Ceil(c.RateControlConfig.EventRate / float64(c.RateControlConfig.Concurrency))
 	}
 
-	if c.RateControl.Concurrency == 0 {
-		c.RateControl.Concurrency = int(math.Ceil(c.RateControl.EventRate / float64(c.RateControl.MaxEventRatePerGoroutine)))
+	if c.RateControlConfig.OuterLoopRate == 0 {
+		c.RateControlConfig.OuterLoopRate = 50
 	}
 
-	if c.Name == "" {
-		panic("must specify Name")
+	if c.WorkloadScale > 0 {
```

Comment:
Do we need to do this check? Why not just multiply this value with the other values and get 0 for everything?

Reply:
As mentioned above, I'm using 0 as the default do-nothing value.

Actually, an earlier version had
```
if c.WorkloadScale == 0 {
        c.WorkloadScale = 1
}
```
right at the top, and then just used 1.0 as the scaling value (and also allowed the user to specify it as 1.0).
Maybe that's better for readability, even with the pointless multiplying by 1.0?